lib/core/utils/parse-sameorigin-stylesheets.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * Parse non cross-origin stylesheets
+ *
+ * @param {Object} sheet CSSStylesheet object
+ * @param {Object} options `loadCssom` options
+ * @param {Array<Number>} priority sheet priority
+ * @param {Array<String>} importedUrls urls of already imported stylesheets
+ * @param {Boolean} isCrossOrigin boolean denoting if a stylesheet is `cross-origin`
+ * @returns {Promise}
+ */
+axe.utils.parseSameOriginStylesheets = function parseSameOriginStylesheets(
+	sheet,
+	options,
+	priority,
+	importedUrls,
+	isCrossOrigin = false
+) {
+	const rules = Array.from(sheet.cssRules);
+
+	if (!rules) {
+		return Promise.resolve();
+	}
+
+	/**
+	 * reference -> https://developer.mozilla.org/en-US/docs/Web/API/CSSRule#Type_constants
+	 */
+	const cssImportRules = rules.filter(r => r.type === 3); // type === 3 -> CSSRule.IMPORT_RULE
+
+	/**
+	 * when no `@import` rules in given sheet -> resolve the current `sheet` & exit
+	 */
+	if (!cssImportRules.length) {
+		// exit
+		return Promise.resolve({
+			isCrossOrigin,
+			priority,
+			root: options.rootNode,
+			shadowId: options.shadowId,
+			sheet
+		});
+	}
+
+	/**
+	 * filter rules that are not already fetched
+	 */
+	const cssImportUrlsNotAlreadyImported = cssImportRules
+		// ensure rule has a href
+		.filter(rule => rule.href)
+		// extract href from object
+		.map(rule => rule.href)
+		// only href that are not already imported
+		.filter(url => !importedUrls.includes(url));
+
+	/**
+	 * iterate `@import` rules and fetch styles
+	 */
+	const promises = cssImportUrlsNotAlreadyImported.map(
+		(importUrl, cssRuleIndex) => {
+			const newPriority = [...priority, cssRuleIndex];
+			const axiosOptions = {
+				method: 'get',
+				url: importUrl,
+				timeout: options.timeout
+			};
+			const isCrossOriginRequest = /^https?:\/\/|^\/\//i.test(importUrl);
+
+			importedUrls.push(importUrl);
+			return axe.imports
```

Comment:
This looks like the same code from `parseCrossOriginStylesheets`. It even includes the same unnecessary Promise-wrapping.

Can we clean this up (and only worry about these LOCs once)?